pa4/checker-pa4.ml

```diff
@@ -610,7 +646,28 @@ let rec exp_typecheck(o_e: obj_env) (m_e: method_env) (exp: expression) : static
 						exit 1
 					end
 		) bindings;
-		exp_typecheck new_o m_e exp
+		exp_typecheck new_o m_e c_e exp
+	| While(exp1, exp2) -> 
```

Comment:
I think that calling these something like `termination_condition_expression` and `body_expression` would make it clearer what these were doing.